### DISCUSSION TITLE
I would like to do something special when ENOCHANGE occurs

### DISCUSSION THREAD

**Discussion by djeanprost** [semantic-release/semantic-release#3391] [Q&A] [answered]
Hello
I sometimes have CI pipeline that run semantic release which does nothign because no commits triggrered any release, which gives this as you know:
`[9:15:37 AM] [semantic-release] › ℹ  There are no relevant changes, so no new version is released.`

Unfortunatly for the moment, my pipeline goes on after this non release step, and it fails in a later job. I would prefer to fail fast and do something when ENOCHANGE occurs.
What is the best way to detect this ? Should I use the `fail` event in semantic release ? is it possible to setup semantic release to exit when no release is produced ?
Thank you

**Comment by travi** [comment, marked as answer]
it is expected that commits will be made that should not result in a release. this is not an exceptional situation, which is why semantic-release exits without error. if there are parts of your pipeline that depend on whether or not semantic-release releases or not, those steps would be best handled by including them as part of the flow that semantic-release is coordinating. that is done through [plugins](https://semantic-release.gitbook.io/semantic-release/usage/plugins). if you dont think your workflow justifies creating a custom plugin, our [official exec plugin](https://github.com/semantic-release/exec) might enable what you need.
